src/function/arithmetic/multiply.js

```diff
@@ -150,12 +150,20 @@ export const createMultiply = /* #__PURE__ */ factory(name, dependencies, ({ typ
       c[j] = sum
     }
 
-    // return matrix
-    return a.createDenseMatrix({
+    // matrix to return
+    const returnMatrix = a.createDenseMatrix({
       data: c,
       size: [bcolumns],
       datatype: dt
     })
+
+    // Set the type to undefined if it wasn't defined for a or b
+    if (adt !== a._datatype || bdt !== b._datatype) {
+      returnMatrix._datatype = undefined
+    }
+
+    // return matrix
+    return returnMatrix
```

Comment:
I think this can be written shorter like:

```js
    return a.createDenseMatrix({
      data: c,
      size: [bcolumns],
      datatype: adt === a._datatype && bdt === b._datatype ? dt : undefined
    })
```

Reply:
I added this in the latest commit